Job board ver2

sophm256 opened the pull request:
These were the following changes done to meet the MVP and a bit extra (#270):

- Add required field for "How to Apply" #301
- Change the external recruiter question to:  "Is this posting from a recruiting agency?" #300
- Make the base URL of the job board: chipy.org/job-board
- Add a link on this page to "Submit Job" that takes us to the create view where labels and form elements are shown side by side
- Add 'Location' field with dropdown options: (Chicago, Chicago and Remote, Remote Only, Chicago and Temporarily Remote) #303
- Add Job Type Field with dropdown options:  (Full-Time, Contract to Hire Full Time, Part Time, Paid Internship, Paid Apprenticeship) #304
- On the 'after_submit_job_post.html', show a list of 'My Jobs' (i.e. the job posts that the user has submitted)
- Add link to 'My Jobs' in the user login dropdown menu #307 
- Add the ability to update a job post
- Add the ability to delete a job post
- Improve page that shows list of posted jobs by using Bootstrap CSS for tables #305 
- Add required field 'How to Apply' #301
- Add 'job board' link to main navbar #310
- Add pagination to 'after_submit_job_post.html' and 'job_post_list.html'

sophm256 commented:
This is what it'll look like at `chipy.org/job-board`. The overall look was improved with Bootstrap CSS for tables.  Also a 'Submit Jobs' button was added to take the user to the createview for the job post. The user can click anywhere on the table row to select the job, or click on the blue links for each job role.
![job-board-list](https://user-images.githubusercontent.com/53249319/88924719-3f963980-d239-11ea-8009-044ab842e12b.JPG)

sophm256 commented:
THE CREATE VIEW FOR JOB POSTS

These first three pictures are part of the same view.

This is the create view for the job post. The recruiting agency verbage has changed to `"Is this posting from a recruiting agency?"` The fields `'Job Type'`,` 'Location'` (with the help message 'ChiPy is a locally based group'), and `'How to Apply'.`

------------------------------------------------------- PART 1 -------------------------------------------------------
![create-job-post](https://user-images.githubusercontent.com/53249319/88925731-90f2f880-d23a-11ea-8f16-efe2c3c51276.JPG)

-------------------------------------------------------PART 2 ---------------------------------------------------------------------------
![create-job-post-2](https://user-images.githubusercontent.com/53249319/88925744-95b7ac80-d23a-11ea-9113-5e72cbefcaec.JPG)
-------------------------------------------------------PART 3 ---------------------------------------------------------------------------
![create-job-post3](https://user-images.githubusercontent.com/53249319/88925755-98b29d00-d23a-11ea-9781-fcc46bb2dafc.JPG)

-----------------------------------------------------------------------------------------------------------------------------------
LOCATION REQUIREMENTS MESSAGE

When you click on the `'more'` teal button that's located next to the 'location' field, a modal pops up with a message.
![location-help-msg](https://user-images.githubusercontent.com/53249319/88926461-a4eb2a00-d23b-11ea-8d98-429d897a6d3e.JPG)

sophm256 commented:
THE AFTER_SUBMIT_JOB_POST PAGE (which shows a list of 'My Jobs')

AFTER A SUCCESSFUL JOB POST SUBMISSION

- Using Django's message system, the message saying the job post was successfully submitted is shown in green box.
- User can submit more job posting by clicking on 'Submit More Jobs' which takes them to the create view page
- User can update or delete post
- User can see the status

![after-submitting](https://user-images.githubusercontent.com/53249319/88927022-6f930c00-d23c-11ea-8a51-134faddf3ca3.JPG)

sophm256 commented:
UPDATING A JOB POST

The page to update the job post is basically the same as the create view, except for the title and the 'Update Post' button at the end. 

![update-post](https://user-images.githubusercontent.com/53249319/88928064-f72d4a80-d23d-11ea-8a0f-b164abf79431.JPG)

--------------------------------------------------------------------------------------------------

Here is the success message (in the green box) after updating a post.

![after-update](https://user-images.githubusercontent.com/53249319/88928221-2cd23380-d23e-11ea-9eef-499a55c10aff.JPG)

sophm256 commented:
DELETING A POST

After the user clicks on the 'Delete' button to delete a post, the user will be taken to a confirmation view. They'll be asked if they're sure they want to delete it.

![delete-confirmation](https://user-images.githubusercontent.com/53249319/88928687-b84bc480-d23e-11ea-98d0-7565715c102d.JPG)

----------------------------------------------

This is the success message (in the green box) after deleting the post.

![after-delete-msg](https://user-images.githubusercontent.com/53249319/88928910-fe088d00-d23e-11ea-81b2-f64f3be75813.JPG)

sophm256 commented:
MORE INFO REGARDING THE 'AFTER_SUBMIT_JOB_POST' PAGE

The status of the job post can be : 'Submitted', 'Approved, and 'Rejected. 
- If it's in the 'submitted' state, they can update or delete the post.
- If it's in the 'approved' state, they can view the job post 'detail' page (which is how each singular job post will show up) or delete it.
- If it's in the 'rejected' state, they can click on the 'more info' button, which will give them a rejection message. The user can't delete the post. This is in case the user has questions and wants to ask the admin why they have rejected the post. It also leaves open the possibility that the post can be edited in the future so that it can suitably shown.
 
![three-states](https://user-images.githubusercontent.com/53249319/88929024-2abca480-d23f-11ea-92bb-ad1a3f9269a2.JPG)
-------------------------------------------------------------------------------

REJECTION MESSAGE

Here's rejection message when user clicks the teal 'More Info' button.

![rejection-message](https://user-images.githubusercontent.com/53249319/88929709-1af19000-d240-11ea-921f-95883d0cf8a6.JPG)

-------------------------------------------------------------------------------

DETAIL PAGE FOR A SINGLE JOB POST

![job-detail](https://user-images.githubusercontent.com/53249319/88929896-5b510e00-d240-11ea-9d63-48bc9a8cf5d7.JPG)

sophm256 commented:
MORE INFO REGARDING THE 'AFTER_SUBMIT_JOB_POST' PAGE

The status of the job post can be : 'Submitted', 'Approved, or 'Rejected. 
- If it's in the 'submitted' state, they can update or delete the post.
- If it's in the 'approved' state, they can view the job post 'detail' page (which is how each singular job post will show up) or delete it.
- If it's in the 'rejected' state, they can click on the 'more info' button, which will give them a rejection message. The user can't delete the post. This is in case the user has questions and wants to ask the admin why they have rejected the post. It also leaves open the possibility that the post can be edited in the future so that it can be suitably shown.
 
-Also a link to this page is shown in the upper right, in the user login dropdown menu. When the user clicks on `'My Jobs',` it'll take them to this `'after_submit_job_post' page`.

![three-states-dropdown-2](https://user-images.githubusercontent.com/53249319/88946864-619db500-d255-11ea-870d-8b80b68bd74c.jpg)


-------------------------------------------------------------------------------

REJECTION MESSAGE

Here's rejection message when user clicks the teal 'More Info' button.

![rejection-message](https://user-images.githubusercontent.com/53249319/88946900-6bbfb380-d255-11ea-9963-00131aef9f10.JPG)

-------------------------------------------------------------------------------

DETAIL PAGE FOR A SINGLE JOB POST

![job-detail](https://user-images.githubusercontent.com/53249319/88947029-96aa0780-d255-11ea-9b20-4a02e45f952a.JPG)

sophm256 commented:
ADDED 'JOB BOARD' LINK TO MAIN NAVBAR (#310)

![job-board-navbar](https://user-images.githubusercontent.com/53249319/89096304-46d55880-d39b-11ea-82ba-83486f15989d.JPG)

sophm256 commented:
ADDED 'JOB BOARD' LINK TO MAIN NAVBAR (#310)

![job-board-navbar-fixed](https://user-images.githubusercontent.com/53249319/89096760-86ea0a80-d39e-11ea-9722-280674c17b67.JPG)

elmq0022 commented:
I think you should run a quick rebase on master.  I pulled your code, and I see a couple of the recent bug fixes haven't made it into this branch.

elmq0022 commented:
This is great work and I believe it meets our requirements for the MVP.  I did have a couple thoughts as went through this.  Let me know if you want me enter a pull request for any of them. 

![chipy-local-text](https://user-images.githubusercontent.com/3450848/89114688-f9adc100-d444-11ea-8627-25d35552aad7.png)

Maybe this could match the current flap :

> Position must not move candidate out of the Chicago area. Working remote or commuting is acceptable. Any position requiring relocation out of the Chicago land is out of scope of the Chipy's mission.

![dead-referral-terms-link](https://user-images.githubusercontent.com/3450848/89114749-a8520180-d445-11ea-9e0d-8520dd297910.png)

Glad to see the flat page already exists on chipy.org

![status-drop-down-not-approve-reject-button](https://user-images.githubusercontent.com/3450848/89114876-a63c7280-d446-11ea-8fa7-ac2379ac495e.png)
 
This actually took me a minute to find.  Would like an approve or reject box, but admins can just deal with it as is I think.  

![my-jobs-hidden-in-drop-down](https://user-images.githubusercontent.com/3450848/89114899-cbc97c00-d446-11ea-95e7-152a797b2bd0.png)

You have good instructions around this, but I like many other users don't really read instructions.  Coule we maybe have a button that's visible at the to of the page or something not hidden in a hamburger menu?  I think this would be easier for the user.

![should-this-edit-the-users-first-and-last-name](https://user-images.githubusercontent.com/3450848/89114916-0f23ea80-d447-11ea-87d4-38bf6b4633e3.png)

These fields let's the user update their actual first and last name tied to their user profile.  Maybe this could be improved to pre-populate that information if it already exists? 

![company-website-is-not-a-link](https://user-images.githubusercontent.com/3450848/89114940-5a3dfd80-d447-11ea-81ef-16fb23cebeb0.png)

I would like this to be an actual link if possible.

![could-edit-put-back-to-submitted-status](https://user-images.githubusercontent.com/3450848/89114947-6f1a9100-d447-11ea-99a8-deed6e7a89cb.png)

maybe in a future release we leave the update button, and the status is just toggled back to submitted

![this-looks-odd-without-a-sponsored-job-post](https://user-images.githubusercontent.com/3450848/89114955-8e192300-d447-11ea-8048-2aabcb0c63ac.png)

Could we make this just Sponsored Jobs and Jobs instead of other jobs?  Seems odd if it's only other jobs on the page.  Or we could just have both headers and no jobs listed under them if they don't exist.  

Again super.   :beers:

sophm256 commented:
ADDED ERROR FIELDS IN 'AFTER_SUBMIT_JOB_POST' PAGE

![error-messages](https://user-images.githubusercontent.com/53249319/89114984-1c8da480-d448-11ea-8f06-b6b3f3844d67.JPG)

............................................................................................................................................................................



ADDED PAGINATION TO 'AFTER_SUBMIT_JOB_POST' PAGE AND THE MAIN JOB BOARD PAGE **(It is currently aligned to the left of the page and will probably need to be centered in the page in the future.)**

![pagination](https://user-images.githubusercontent.com/53249319/89114993-42b34480-d448-11ea-8c14-de0bbcf4731e.JPG)

elmq0022 commented:
Also I notice that we can put negative values for the days to show posts.  

![image](https://user-images.githubusercontent.com/3450848/89115147-45af3480-d44a-11ea-9b40-04358770d3f4.png)

Was a nice way to test the functionality though :smile:

sophm256 commented:
@elmq0022 Thanks a lot for your review. The feedback was really helpful. I made the most of changes you suggested. 

----------------------------CHANGES MADE--------------------------

 CHANGED THE LOCATION REQUIREMENTS MESSAGE (Heather had wanted "ChiPy is a locally based group" to be included.  But the rest of the message changed to be the same as from the referral flatpage.)

![location-requirements-update](https://user-images.githubusercontent.com/53249319/89117343-9599f580-d462-11ea-9d67-ff7f5d915bdd.JPG)
.......................................................................................................................................

- MOVED 'MY JOBS' LINK FROM DROPDOWN MENU TO  MAIN MENU
- CHANGED 'OTHER JOB POSTS' TO 'JOB POSTS'
- CHANGED TEAL BLUE TITLE 'JOB POSTS' TO 'JOB BOARD'

![job-boar-main-updated](https://user-images.githubusercontent.com/53249319/89117358-bd895900-d462-11ea-9d62-12e81d6686f6.JPG)
.......................................................................................................................................

- CHANGED SUCCESS MESSAGE IN GREEN SO THAT USERS CAN FIND 'MY JOBS' LINK IN MAIN MENU
- CHANGED TITLE 'MY JOB POSTS' TO 'MY JOBS'

![after-submit-success-my-jobs](https://user-images.githubusercontent.com/53249319/89117418-42747280-d463-11ea-9843-d84b53f7064c.JPG)

sophm256 commented:
@elmq0022, My additional thoughts:

1. In regards to your comment, 

> These fields let's the user update their actual first and last name tied to their user profile. Maybe this could be improved to pre-populate that information if it already exists?


The first name, last name, and email fields are prepopulated if they exist, because it's pulling if from the User class.

.......................................................................................................................................

2. 

> I would like this to be an actual link if possible. I'm not sure what you mean. 

Do you mean you'd like the company website, such as google.com, to  show up as a clickable link?

.......................................................................................................................................

3. 

> maybe in a future release we leave the update button, and the status is just toggled back to submitted .

I don't understand this comment. Could you please explain it further?

 .......................................................................................................................................


4. Yes, it's pretty neat that you can put a negative number for the days to show post. This allows one to disable the post from showing up on the job board without deleting it. 

.......................................................................................................................................

5.  **If would be really helpful if you could write those tests. Thanks again for all your help!** :cake:

raymondberg commented on chipy_org/apps/job_board/templates/job_board/job_post_form.html:
```diff
@@ -0,0 +1,113 @@
+        {% endfor %}
+     
+        {% for field in job_post_form %}
+            {% if field == job_post_form.agree_to_terms %}
```
It'd be great to figure out how to avoid this if breakout, but for now this makes a lot of sense with the workarounds needed on several fields.

sophm256 replied:
Agreed. :+1:

raymondberg commented:
Merged with 6318207ca3ced44b52e93c7e44973fa5ef52b417
